src/ripple/protocol/STObject.h

```diff
@@ -970,6 +971,27 @@ STObject::operator[](OptionaledField<T> const& of) ->
     return OptionalProxy<T>(this, of.f);
 }
 
+inline
+NetClock::time_point
+getsfCloseTime(STObject const& sle)
```

Comment:
The rabbit hole was too deep.  I got things to compile and introduced a tough-to-track-down run time error (lack of type safety due to the need to serialize).  It can be done, but it is too big and risky of a change to put into this already big P/R.  I'm happy to put it into a separate P/R.  In the mean-time I've removed these helper functions.